src/basic/log.c

```diff
@@ -607,15 +625,19 @@ static int write_to_journal(
 
         log_do_header(header, sizeof(header), level, error, file, line, func, object_field, object, extra_field, extra);
 
-        struct iovec iovec[4] = {
+        struct iovec iovec[IOVEC_MAX] = {
```

Comment:
that's quite an increase. i.e. 32 byte before. 1K now. in a world where 4K stacks are often used... For a log function that should try to be as little intrusive as we can be.

maybe count how much we actually need, and size the thing properly?

Reply:
Added counting, but kept the upper limit just in case